homeassistant/components/broadlink/switch.py

```diff
@@ -43,80 +45,59 @@
     }
 )
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Optional(CONF_HOST): cv.string,
-        vol.Optional(CONF_MAC): mac_address,
-        vol.Optional(CONF_SWITCHES, default=[]): vol.Any(
-            cv.schema_with_slug_keys(OLD_SWITCH_SCHEMA),
-            vol.All([SWITCH_SCHEMA], cv.ensure_list),
-        ),
-        vol.Optional(CONF_TYPE): cv.string,
-        vol.Optional(CONF_TIMEOUT): cv.positive_int,
-    },
-    extra=vol.ALLOW_EXTRA,
+PLATFORM_SCHEMA = vol.All(
+    cv.deprecated(CONF_HOST),
+    cv.deprecated(CONF_SLOTS),
+    cv.deprecated(CONF_TIMEOUT),
+    cv.deprecated(CONF_TYPE),
+    PLATFORM_SCHEMA.extend(
+        {
+            vol.Required(CONF_MAC): mac_address,
+            vol.Optional(CONF_HOST): cv.string,
+            vol.Optional(CONF_SWITCHES, default=[]): vol.Any(
+                cv.schema_with_slug_keys(OLD_SWITCH_SCHEMA),
+                vol.All(cv.ensure_list, [SWITCH_SCHEMA]),
+            ),
+        },
+        extra=vol.ALLOW_EXTRA,
```

Comment:
Since we're still using this schema, it's better to not allow extra and keep the deprecated options in the schema until it's time to remove them. We don't want to allow arbitrary keys in the schema. That can lead to confusion for users.

Reply:
Ok.